File: backend/app/api/v1/endpoints/pipelines.py
from typing import List
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession
from app.core.database import get_db
from app.core.deps import get_current_active_user
from app.models.user import User

router = APIRouter()


@router.get("/")
async def get_pipelines(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现流水线查询
    return []


@router.get("/{pipeline_id}")
async def get_pipeline(
    pipeline_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现单个流水线查询
    return {}


@router.post("/")
async def create_pipeline(
    pipeline_data: dict,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现流水线创建
    return pipeline_data


@router.put("/{pipeline_id}")
async def update_pipeline(
    pipeline_id: str,
    pipeline_data: dict,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现流水线更新
    return pipeline_data


@router.delete("/{pipeline_id}")
async def delete_pipeline(
    pipeline_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现流水线删除
    return {"message": "Pipeline deleted"}


@router.post("/{pipeline_id}/run")
async def run_pipeline(
    pipeline_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现流水线运行
    return {
        "id": "run-001",
        "pipeline_id": pipeline_id,
        "status": "running",
        "trigger_type": "manual",
        "started_at": "2024-01-01T00:00:00Z",
        "stages": []
    }


@router.get("/{pipeline_id}/runs")
async def get_pipeline_runs(
    pipeline_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现运行历史查询
    return []


@router.get("/{pipeline_id}/runs/{run_id}")
async def get_pipeline_run(
    pipeline_id: str,
    run_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    # TODO: 实现单次运行详情查询
    return {}

File: frontend/src/api/pipelines.ts
import request from '@/utils/request'

export interface Pipeline {
  id: string
  name: string
  description?: string
  repo_id: string
  repo_name?: string
  branch: string
  trigger_type: string
  status: string
  stages: PipelineStage[]
  is_active: boolean
  last_run_at?: string
  last_run_status?: string
  created_by?: string
  created_at: string
  updated_at: string
}

export interface PipelineStage {
  name: string
  order: number
  script_id?: string
  script_name?: string
  status?: string
}

export interface PipelineCreate {
  name: string
  description?: string
  repo_id: string
  branch?: string
  trigger_type: string
  stages: Omit<PipelineStage, 'status'>[]
}

export interface PipelineUpdate {
  name?: string
  description?: string
  repo_id?: string
  branch?: string
  trigger_type?: string
  stages?: Omit<PipelineStage, 'status'>[]
  is_active?: boolean
}

export interface PipelineRun {
  id: string
  pipeline_id: string
  status: string
  trigger_type: string
  started_at: string
  finished_at?: string
  duration?: number
  stages: PipelineRunStage[]
}

export interface PipelineRunStage {
  name: string
  status: string
  started_at?: string
  finished_at?: string
  output?: string
}

export const pipelinesApi = {
  // Get all pipelines
  getPipelines() {
    return request.get<Pipeline[]>('/v1/pipelines/')
  },

  // Get pipeline by ID
  getPipeline(pipelineId: string) {
    return request.get<Pipeline>(`/v1/pipelines/${pipelineId}`)
  },

  // Create pipeline
  createPipeline(data: PipelineCreate) {
    return request.post<Pipeline>('/v1/pipelines/', data)
  },

  // Update pipeline
  updatePipeline(pipelineId: string, data: PipelineUpdate) {
    return request.put<Pipeline>(`/v1/pipelines/${pipelineId}`, data)
  },

  // Delete pipeline
  deletePipeline(pipelineId: string) {
    return request.delete<{ message: string }>(`/v1/pipelines/${pipelineId}`)
  },

  // Run pipeline
  runPipeline(pipelineId: string) {
    return request.post<PipelineRun>(`/v1/pipelines/${pipelineId}/run`)
  },

  // Get pipeline runs
  getPipelineRuns(pipelineId: string) {
    return request.get<PipelineRun[]>(`/v1/pipelines/${pipelineId}/runs`)
  },

  // Get pipeline run details
  getPipelineRun(pipelineId: string, runId: string) {
    return request.get<PipelineRun>(`/v1/pipelines/${pipelineId}/runs/${runId}`)
  },
}
